src/url.cpp

```diff
@@ -298,4 +305,76 @@ namespace Url
         return *this;
     }
 
+    Url& Url::relative(const std::string& other)
+    {
+        if (other.empty())
+        {
+            return *this;
+        }
+
+        Url parsed(other);
+
+        if (!parsed.host_.empty())
+        {
+            // Support scheme-relative URLs
+            if (!parsed.scheme_.empty())
+            {
+                scheme_ = parsed.scheme_;
+            }
+
+            host_ = parsed.host_;
+            port_ = parsed.port_;
+            path_ = parsed.path_;
+            params_ = parsed.params_;
+            query_ = parsed.query_;
+            fragment_ = parsed.fragment_;
+            userinfo_ = parsed.userinfo_;
+            return *this;
+        }
+
+        if (!parsed.path_.empty() && parsed.path_.front() == '/')
+        {
+            path_ = parsed.path_;
+            params_ = parsed.params_;
+            query_ = parsed.query_;
+            fragment_ = parsed.fragment_;
+            return *this;
+        }
+
+        if (parsed.path_.empty() && parsed.params_.empty())
+        {
+            if (!parsed.query_.empty())
+            {
+                query_ = parsed.query_;
+            }
+            fragment_ = parsed.fragment_;
+            return *this;
+        }
+
+        // Copy the parsed path onto the end of the last segment of this path
+        size_t index = path_.rfind('/');
+        if (index == std::string::npos)
+        {
+            if (path_.empty())
+            {
+                path_ = "/" + parsed.path_;
```

Comment:
I'm not sure I follow why this leading `/` is being added. It seems like either `parsed.path` would have a leading slash in which case you wouldn't prepend one or it is empty, in which case it's handled above, or it's a relative path in which case you really don't want to prepend a `/` to it.